### PR TITLE
feat(cli): Add json import/export of app config

### PR DESCRIPTION
**Pull request by darkliquid**
Add the capability to export and import the application config to/from a json file. The JSON file is LocalState\config.json in the app package folder. The options are available via the new flags to the settings command, e.g. `fluent.exe settings --export` and `fluent.exe settings --import`

Import attempts to merge imported config with existing config, and refuses to import thigns marked as pre-installed. It attempts to merge configuration for pre-installed profiles where those fields are editable, but otherwise discards invalid configuration.

Addresses: https://github.com/felixse/FluentTerminal/issues/283

### CODE REVIEW

File: FluentTerminal.App/App.xaml.cs [hunk 1]
```diff
@@ -160,9 +160,72 @@ protected override void OnActivated(IActivatedEventArgs args)
...
+                            var stream = await exportFile.OpenStreamForWriteAsync();
+                            JsonSerializer s = new JsonSerializer();
+                            using (StreamWriter sw = new StreamWriter(stream))
+                            using (JsonWriter writer = new JsonTextWriter(sw))
+                            {
+                                var config = new
+                                {
+                                    App = _settingsService.GetApplicationSettings(),
+                                    KeyBindings = _settingsService.GetCommandKeyBindings(),
+                                    Themes = _settingsService.GetThemes(),
```
Comment (felixse) [inline FluentTerminal.App/App.xaml.cs]: We should avoid exporting preinstalled themes and profiles

File: FluentTerminal.App/App.xaml.cs [hunk 2]
```diff
@@ -160,9 +160,72 @@ protected override void OnActivated(IActivatedEventArgs args)
...
+                                    Profiles = _settingsService.GetShellProfiles(),
+                                    TerminalOptions = _settingsService.GetTerminalOptions(),
+                                    TabThemes = _settingsService.GetTabThemes()
```
Comment (felixse) [inline FluentTerminal.App/App.xaml.cs]: TabThemes are static for now, we don't need to export/import them

File: FluentTerminal.App/App.xaml.cs [hunk 3]
```diff
@@ -160,9 +160,72 @@ protected override void OnActivated(IActivatedEventArgs args)
...
+                                _settingsService.SaveApplicationSettings(config.App);
+                                
+                                foreach (var pair in config.KeyBindings)
+                                {
+                                    _settingsService.SaveKeyBindings(pair.Key, pair.Value);
+                                }
+
+                                foreach (var theme in config.Themes)
+                                {
+                                    _settingsService.SaveTheme(theme, _settingsService.GetTheme(theme.Id).Equals(default(TerminalTheme)));
```
Comment (felixse) [inline FluentTerminal.App/App.xaml.cs]: If we only export non preinstalled themes/profiles we can hardcore IsNew to true

**Comment by darkliquid** [comment]
Pushed an update that hopefully makes a bit more sense - it refuses to operate on `Preinstalled=true` themes and profiles now for both export and import. I'm also trying to make it a little friendlier to merging with existing config, rather than completely replacing it with a fusion of default settings and whatever you happen to have imported.

**Comment by felixse** [comment]
Thanks 👍 
I ahve moved the logic to the settings and fixed some issues. Next I'm gonna write some tests and then this can get merged

**Comment by darkliquid** [comment]
Nice!

**Pull request merged**